applications/GeoMechanicsApplication/custom_processes/set_parameter_field_process.cpp

```diff
@@ -0,0 +1,143 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aron Noordam
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "set_parameter_field_process.hpp"
+
+#include <utilities/function_parser_utility.h>
+#include <utilities/mortar_utilities.h>
+
+#include "utilities/interval_utility.h"
+#include "geo_mechanics_application_variables.h"
+
+
+namespace Kratos
+{
+
+    SetParameterFieldProcess::SetParameterFieldProcess(ModelPart& rModelPart,
+                                                            Parameters Settings)
+                                                            : mrModelPart(rModelPart),
+                                                            mParameters(Settings)
+{
+    // function type: python, cpp, input
+    Parameters default_parameters(R"(
+        {
+            "help"            : "This process applies a moving load condition belonging to a modelpart. The load moves over line elements.",
+            "model_part_name" : "please_specify_model_part_name",
+            "variable_name"   : "CUSTOM",
+            "func_type"       : "input",               
+            "function"        : "0",
+            "dataset"         : "dummy"
+        }  )"
+    );
+    Parameters mParameters;
+
+    mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+}
+
+
+void SetParameterFieldProcess::SetValueAtElement(Element& rElement, const Variable<double>& rVar, double Value)
+{
+
+
+    Properties& r_prop = rElement.GetProperties();
+
+    // Copies properties
+    Properties::Pointer p_new_prop = Kratos::make_shared<Properties>(r_prop);
+
+    // Adds new properties to the element
+    p_new_prop->SetValue(rVar, Value);
+    rElement.SetProperties(p_new_prop);
+
+}
+
+
+void SetParameterFieldProcess::ExecuteInitialize()
+{
+    KRATOS_TRY
+
+    if (!this->mrModelPart.GetProcessInfo()[IS_RESTARTED]){
+
+        const Variable<double>& r_var = KratosComponents< Variable<double> >::Get(mParameters["variable_name"].GetString());
+
+        // set parameter field from input function
+        if (mParameters["func_type"].GetString().compare("input") == 0)
```

Comment:
Is there any particular reason why we should use `compare` rather than `operator==`? I feel that the latter is a little easier  to read:
```suggestion
        if (mParameters["func_type"].GetString() == "input")
```

This comment also applies to the other two string comparisons further down.